tests/run-make/rustdoc-test-builder/rmake.rs

```diff
@@ -19,4 +22,38 @@ fn main() {
     output.assert_stdout_contains("Failed to spawn ");
     // ... and that we didn't panic.
     output.assert_not_ice();
+
+    // Some targets (for example wasm) cannot execute doctests directly even with a runner,
+    // so only exercise the success path when the target can run on the host.
+    let target = std::env::var("TARGET").expect("TARGET must be set");
+    if target.contains("wasm") {
+        return;
+    }
```

Comment:
Suggestion: sorry I didn't look at this, but this should instead use `//@ ignore-wasm` or `//@ ignore-cross-compile` (if this is host-only), otherwise this will show as test pass on WASM but in reality nothing is checked. Since this is in a rollup, it can be done as a follow-up.

(Also, there is a `run_make_support::target()` helper for what you're doing here.)